### PR TITLE
PEP 621: Migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by cclauss**
https://setuptools.pypa.io/en/latest/userguide/quickstart.html#basic-use
> We recommend users expose as much as possible configuration in a more declarative way via the [pyproject.toml](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html) or [setup.cfg](https://setuptools.pypa.io/en/latest/userguide/declarative_config.html), and keep the setup.py minimal with only the dynamic parts (or even omit it completely if applicable).

1. Migrate from `setup.py` to `setup.cfg` using [`setuptools-py2cfg`](https://pypi.org/project/setuptools-py2cfg) plus manual modifications.
2. Migrate to `pyproject.toml` using [`ini2toml`](https://pypi.org/project/ini2toml) to do the file conversion
3. Running [`validate-pyproject`](https://github.com/abravalheri/validate-pyproject) to validate the results.
4. Add `validate-pyproject` to tox.

From `validate-pyproject` README:
> With the approval of [PEP 517](https://www.python.org/dev/peps/pep-0517/) and [PEP 518](https://www.python.org/dev/peps/pep-0518/), the Python community shifted towards a strong focus on standardization for packaging software, which allows more freedom when choosing tools during development and make sure packages created using different technologies can interoperate without the need for custom installation procedures.

> This shift became even more clear when [PEP 621](https://www.python.org/dev/peps/pep-0621/) was also approved, as a standardised way of specifying project metadata and dependencies.

> `validate-pyproject` was born in this context, with the mission of validating `pyproject.toml` files, and making sure they are compliant with the standards and PEPs.

### CODE REVIEW

**Comment by melroy89** [comment]
Thanks this was indeed on the planning. Thanks for the help.

**Comment by melroy89** [comment]
@cclauss Could you maybe put the validation check directly below the black test? So before the run of the tests, meaning it should fail faster that way. Since the test cases take the most time.

Fail fast, learn fast and therefore github actions will also stop the matrix runs faster in case of an error.

File: tox.ini [hunk 1]
```diff
@@ -14,8 +15,10 @@ deps =
     isort
     pytest
     pytest-cov
+    validate-pyproject
 commands =
     black --check --diff .
     # Isort needs to be fixed (skip is not recognized anymore)
     # isort --check-only -rc fake_useragent --diff
     pytest tests
+    validate-pyproject pyproject.toml
```
Comment (melroy89) [inline tox.ini:23]: Put the validation above the pytest (also don't forgot the update from upstream), since we have file overlap.

**Review by melroy89 — CHANGES_REQUESTED** [review]
Update the order in tox and update from upstream master branch.

**Comment by cclauss** [comment]
🦮… back soon.

**Comment by cclauss** [comment]
Rebased

**Review by melroy89 — APPROVED** [review]

**Comment by melroy89** [comment]
I think I will move the `version=` to pyproject.toml and use the following in the code:

```py
import importlib.metadata
__version__ = importlib.metadata.version("fake-useragent")
```

What are your thoughts, this is a good practice right?

**Comment by melroy89** [comment]
See follow-up: https://github.com/fake-useragent/fake-useragent/pull/147

**Pull request merged**